source/extensions/filters/network/dubbo_proxy/deserializer.h

```diff
@@ -107,6 +107,18 @@ class Deserializer {
    * @throws EnvoyException if the data is not valid for this serialization
    */
   virtual RpcResultPtr deserializeRpcResult(Buffer::Instance& buffer, size_t body_size) PURE;
+
+  /**
+   * serialize result of an rpc call
+   * If successful, the output_buffer is written to the serialized data
+   *
+   * @param output_buffer store the serialized data
+   * @param content the rpc response content
+   * @param type the rpc response type
+   * @return size_t the length of the serialized content
+   */
+  virtual size_t serializeRpcResult(Buffer::Instance& output_buffer, const std::string& content,
```

Comment:
Is it necessary to return the serialized size? Take the buffer size before serialization minus the serialized buffer size.

Reply:
That's fine, but it's much more user-friendly and convenient for the user to explicitly return the serialization length.